superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx

```diff
@@ -37,38 +38,15 @@ interface ActionButtonsProps {
   dataMaskSelected: DataMaskState;
   dataMaskApplied: DataMaskStateWithId;
   isApplyDisabled: boolean;
+  orientation?: FilterBarLocation;
 }
 
-const ActionButtonsContainer = styled.div<{ width: number }>`
-  ${({ theme, width }) => css`
+const ActionButtonsContainer = styled.div<{
```

Comment:
Totally optional, but for these cases, I like to use [Emotion's Composition](https://emotion.sh/docs/composition).

You can write the individual styles:

```
const containerStyle = (theme: SupersetTheme) => css`
  display: flex;

  && > .filter-clear-all-button {
    color: ${theme.colors.grayscale.base};
    margin-left: 0;
    &:hover {
      color: ${theme.colors.primary.dark1};
    }

    &[disabled],
    &[disabled]:hover {
      color: ${theme.colors.grayscale.light1};
    }
  }
`;

const verticalStyle = (theme: SupersetTheme, width: number) => css`
  flex-direction: column;
  align-items: center;
  pointer-events: none;
  position: fixed;
  z-index: 100;

  // filter bar width minus 1px for border
  width: ${width - 1}px;
  bottom: 0;

  padding: ${theme.gridUnit * 4}px;
  padding-top: ${theme.gridUnit * 6}px;

  background: linear-gradient(
    ${rgba(theme.colors.grayscale.light5, 0)},
    ${theme.colors.grayscale.light5} ${theme.opacity.mediumLight}
  );

  & > button {
    pointer-events: auto;
  }

  & > .filter-apply-button {
    margin-bottom: ${theme.gridUnit * 3}px;
  }
`;

const horizontalStyle = (theme: SupersetTheme) => css`
  margin: 0 ${theme.gridUnit * 2}px;
  && > .filter-clear-all-button {
    text-transform: capitalize;
    font-weight: ${theme.typography.weights.normal};
  }
  & > .filter-apply-button {
    &[disabled],
    &[disabled]:hover {
      color: ${theme.colors.grayscale.light1};
      background: ${theme.colors.grayscale.light3};
    }
  }
`;
```

And apply them conditionally:

```
return (
    <div
      data-test="filterbar-action-buttons"
      css={(theme: SupersetTheme) => [
        containerStyle(theme),
        isVertical ? verticalStyle(theme, width) : horizontalStyle(theme),
      ]}
    >
      ...
    </div>
  );
};
```